File: shoring/compat/src/main/java/com/panopset/compat/Padop.java
package com.panopset.compat;

import java.io.StringWriter;

public class Padop {

  private Padop() {}

  public static String rightPad(String inp, int width) {
    return rightPad(inp, width, ' ');
  }

  public static String rightPad(String inp, int width, char padding) {
    if (inp == null) {
      return "";
    }
    StringWriter sw = new StringWriter();
    sw.append(inp);
    int i = inp.length();
    while (i++ < width) {
      sw.append(padding);
    }
    return sw.toString();
  }

  public static String leftPad(String inp, int width) {
    return leftPad(inp, width, ' ');
  }

  public static String leftPad(String inp, int width, char padding) {
    if (inp == null) {
      return "";
    }
    StringWriter sw = new StringWriter();
    int i = inp.length();
    while (i++ < width) {
      sw.append(padding);
    }
    sw.append(inp);
    return sw.toString();
  }
}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/Padop.kt
package com.panopset.compat

import java.io.StringWriter

object Padop {
    fun rightPad(inp: String?, width: Int, padding: Char = ' '): String {
        if (inp == null) {
            return ""
        }
        val sw = StringWriter()
        sw.append(inp)
        var i = inp.length
        while (i++ < width) {
            sw.append(padding)
        }
        return sw.toString()
    }

    fun leftPad(inp: String?, width: Int, padding: Char = ' '): String {
        if (inp == null) {
            return ""
        }
        val sw = StringWriter()
        var i = inp.length
        while (i++ < width) {
            sw.append(padding)
        }
        sw.append(inp)
        return sw.toString()
    }
}
